### DISCUSSION TITLE
Adjust the Counter page in the Example, to jump multiple levels up

### DISCUSSION THREAD

**Discussion by palkatap** [arkivanov/Decompose#473] [Q&A]
I'm just started to check this library, so I'm not that familiar with it
I would like to write a small program that has similar functionality as you Counters page. However I would like to extend the top app bar to include a button for each parent. Clicking on that button returns me to the corresponding level.
How should I start with this?
Could you eventually adjust your example to have this feature, please?

**Comment by arkivanov** [comment]
Thanks for the question! It's not clear how exactly that button should work. There is back button in the [Counter sample](https://github.com/arkivanov/Decompose/blob/e5afc98288fc925774d68a238e13e534d2b9ac7c/sample/shared/compose/src/commonMain/kotlin/com/arkivanov/sample/shared/counters/counter/CounterContent.kt#L36) that switches to the previous Counter in the stack.

**Comment by palkatap** [comment]
I was thinking of something like:
![image](https://github.com/arkivanov/Decompose/assets/25684680/1358e749-3274-4220-a0d0-4c3692f430ca)
For example, clicking on the "Counter 0" button would take me to the topmost level.

**Reply by arkivanov** [reply]
There are multiple options. For example, you can pass a list of button titles to every child Composable function, and a click listener. In the parent Composable, when the listener is called, you can call [counter.onBackClicked(toIndex)](https://github.com/arkivanov/Decompose/blob/e5afc98288fc925774d68a238e13e534d2b9ac7c/sample/shared/shared/src/commonMain/kotlin/com/arkivanov/sample/shared/counters/CountersComponent.kt#L11).

Let me know if you need further assistance with this.
